modules/siddhi-core/src/main/java/io/siddhi/core/table/record/AbstractQueryableRecordTable.java

```diff
@@ -728,9 +721,26 @@ public StreamEvent query(StateEvent matchingEvent, CompiledCondition compiledCon
                 streamEventComplexEventChunk.add(streamEvent);
             }
         }
-        return streamEventComplexEventChunk.getFirst();
     }
 
+    private StreamEvent getStreamEvent(Attribute[] outputAttributes,
+                                       ComplexEventChunk<StreamEvent> streamEventComplexEventChunk,
+                                       CompiledSelectionWithCache compiledSelectionWithCache,
+                                       StreamEvent cacheResults) {
+        StateEventFactory stateEventFactory = new StateEventFactory(compiledSelectionWithCache.
+                metaStateEvent);
+        Event[] cacheResultsAfterSelection = executeSelector(cacheResults,
```

Comment:
Won't the executeSelector give a set of coned/new events? do we need to create another new instance?

Reply:
AQRT Query api need to return StreamEvent where execute selector gives Event[]

Reply:
done